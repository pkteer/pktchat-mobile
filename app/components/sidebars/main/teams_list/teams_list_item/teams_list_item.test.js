// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import Preferences from '@mm-redux/constants/preferences';

import TeamsListItem from './teams_list_item';

describe('TeamsListItem', () => {
    const baseProps = {
        selectTeam: jest.fn(),
        testID: 'main.sidebar.teams_list.flat_list.teams_list_item',
        currentTeamId: 'current-team-id',
        currentUrl: 'current-url',
        displayName: 'display-name',
        mentionCount: 1,
        name: 'name',
        teamId: 'team-id',
        theme: Preferences.THEMES.denim,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<TeamsListItem {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
