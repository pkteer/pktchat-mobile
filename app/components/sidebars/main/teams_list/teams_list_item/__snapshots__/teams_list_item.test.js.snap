// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamsListItem should match snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 10,
    }
  }
  testID="main.sidebar.teams_list.flat_list.teams_list_item"
>
  <TouchableHighlight
    onPress={[Function]}
    underlayColor="rgba(40,66,123,0.5)"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "marginHorizontal": 16,
          "paddingVertical": 10,
        }
      }
      testID="main.sidebar.teams_list.flat_list.teams_list_item.team-id"
    >
      <View>
        <Connect(TeamIcon)
          styleContainer={
            Object {
              "backgroundColor": "#ffffff",
              "height": 40,
              "width": 40,
            }
          }
          styleText={
            Object {
              "fontSize": 18,
            }
          }
          teamId="team-id"
          testID="main.sidebar.teams_list.flat_list.teams_list_item.team_icon"
        />
        <Badge
          containerStyle={
            Object {
              "borderColor": "#1e325c",
              "borderRadius": 14,
              "borderWidth": 2,
              "position": "absolute",
              "right": -12,
              "top": -10,
            }
          }
          count={1}
          countStyle={
            Object {
              "color": "#1e325c",
              "fontSize": 10,
              "fontWeight": "bold",
            }
          }
          extraPaddingHorizontal={10}
          minHeight={20}
          minWidth={20}
          style={
            Object {
              "backgroundColor": "#ffffff",
              "height": 20,
              "padding": 3,
            }
          }
          testID="main.sidebar.teams_list.flat_list.teams_list_item.badge"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "marginLeft": 10,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 18,
            }
          }
          testID="main.sidebar.teams_list.flat_list.teams_list_item.display_name"
        >
          display-name
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(255,255,255,0.5)",
              "fontSize": 12,
            }
          }
        >
          current-url/name
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</View>
`;
